packages/app-store/routing-forms/lib/selectOptions.ts

```diff
@@ -35,14 +35,28 @@ export const getFieldWithOptions = <T extends Field>(field: T) => {
   }
   return {
     ...field,
-  };
+  } as typeof field & z.BRAND<"FIELD_WITH_OPTIONS">;
 };
 
+export function areSelectOptionsInLegacyFormat(
+  field: Pick<Field, "options"> & z.BRAND<"FIELD_WITH_OPTIONS">
```

Comment:
Using brand to ensure that field have `options` property even for legacy field with `selectText`.